### ISSUE TITLE
running example leads to error

### ISSUE THREAD

**Issue by chrisdecker1201** [annikaliebgott/ImFEATbox#5]
I just did the following steps:
1. Install Matlab R2017a and all needed Toolboxes on Ubuntu 16
2. Clone this git repository
3. Set Paths in Matlab
4. Started example_extractFeatures.m as it is and get the following error:

```Matlab
Error using detectHarrisFeatures (line 56)
Expected I to be real.

Error in HarrisF (line 33)
corners = detectHarrisFeatures(I);

Error in fParseFeatureAlgos>@(x)HarrisF(x,plotflag,N_s_Harris)

Error in fExtractFeatures (line 148)
        feat_curr = hFeat(I);

Error in example_extractFeatures (line 50)
    feat_vector = cat(1,fExtractFeatures(images{iI},[],'all'));
```

I tried to give the function a real I by changing line 33 in HarrisF.m to corners = detectHarrisFeatures(real(I)), but I'm not sure if this is the solution and after that I get the following error:

```Matlab
Error using imdilate
Expected input number 1, IM, to be real.

Error in images.internal.morphop>ParseInputs (line 154)
validateattributes(A, {'numeric' 'logical'}, ...

Error in images.internal.morphop (line 23)
 unpacked_M,mex_method] = ParseInputs(varargin{:});

Error in imdilate (line 135)
B = images.internal.morphop(A,se,'dilate',mfilename,varargin{:});

Error in LoGF (line 74)
snlo_dil = imdilate(snlo,ones(3,3,3));

Error in fParseFeatureAlgos>@(x)LoGF(x,sigma,N_blobs,typeflag)

Error in fExtractFeatures (line 148)
        feat_curr = hFeat(I);

Error in example_extractFeatures (line 50)
    feat_vector = cat(1,fExtractFeatures(images{iI},[],'all'));
```

**Comment by annikaliebgott** [comment]
Hi Chris, I just noticed this comment, sorry for the late response. Thanks for the detailed error report. Until now, I did not get that error when I was running the example (Ubuntu 16.04, Matlab R2016b and R2013b), but I'll look into it later today or tomorrow.

**Comment by annikaliebgott** [comment]
Hi Chris, 
I avoided getting that error by adding a check for complex values in I and  using only the real part, after that I didn't get any more errors. I also tried it your way (with the corners = detectHarrisFeatures(real(I)) ) and this worked fine, too. I was not able to reproduce the second error, I'm afraid. I'm going to upload the new version of HarrisF.m later today or this week since I'm going to upload newer versions of some algorithms anyway due to the need to fix some minor bugs.
